client/client.go

```diff
@@ -1386,6 +1390,9 @@ func (c *Client) setupNode() error {
 	if _, ok := node.Meta["connect.sidecar_image"]; !ok {
 		node.Meta["connect.sidecar_image"] = defaultConnectSidecarImage
 	}
+	if _, ok := node.Meta["connect.gateway_image"]; !ok {
+		node.Meta["connect.gateway_image"] = defaultConnectGatewayImage
```

Comment:
Is there any reason to allow distinct images for Connect and Gateways at this point? It seems like there are only 2 use cases for the setting at the moment:

1. Pick a version of Envoy that works best with the version of Consul being used.
2. Use a custom built image.

I can't think of why people would intend to deploy different images, so multiple settings might just make it harder to properly customize.

Reply:
The naming is pretty awkward to reuse as well, so this seems fine.